### PR TITLE
Check previous token has valid position data

### LINKED ISSUE #571: Invalid syntax diagnostic range
Langium version: 0.5.0-next.f2b3802
Package name: langium

## Steps To Reproduce

1. Generate the hello-world project
2. Write incomplete text like this:

```
person Aasdf
person Jo

Hello Jo!
Hello 

```

## The current behavior

```
{
  severity: 1,
  range: {
    start: {
      character: NaN,
      line: NaN,
    },
    end: {
      character: NaN,
      line: NaN,
    },
  },
  message: "Expecting token of type 'ID' but found ``.",
  code: "parsing-error",
  source: "hello-world",
}
```

This results in this error message:

```
rejected promise not handled within 1 second: Error: Invalid arguments
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js-pretty.js:45765
stack trace: Error: Invalid arguments
	at new a (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:76:87496)
	at new a (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:76:84692)
	at asRange (hello-world/node_modules/vscode-languageclient/lib/common/protocolConverter.js:140:24)
	at asDiagnostic (hello-world/node_modules/vscode-languageclient/lib/common/protocolConverter.js:70:66)
	at convertBatch (hello-world/node_modules/vscode-languageclient/lib/common/utils/async.js:181:25)
	at Object.map (hello-world/node_modules/vscode-languageclient/lib/common/utils/async.js:190:17)
	at Object.asDiagnostics (hello-world/node_modules/vscode-languageclient/lib/common/protocolConverter.js:60:22)
	at LanguageClient.workDiagnosticQueue (hello-world/node_modules/vscode-languageclient/lib/common/client.js:990:19)
	at Immediate.<anonymous> (hello-world/node_modules/vscode-languageclient/lib/common/client.js:975:85)
	at processImmediate (node:internal/timers:464:21)
	at process.callbackTrampoline (node:internal/async_hooks:130:17)
```

## The expected behavior

Proper ranges should be computed in any case.

### PR DESCRIPTION
**Pull request by dhuebner**
In empty or only hidden token documents there are no valid previous tokens to find.
This PR handles this cases by just setting the diagnostic range to the document start.

Fixes #571

### CODE REVIEW

File: packages/langium/src/validation/document-validator.ts [hunk 1]
```diff
@@ -76,8 +76,14 @@ export class DefaultDocumentValidator implements DocumentValidator {
...
                     const token = (parserError as MismatchedTokenException).previousToken;
-                    const position = Position.create(token.endLine! - 1, token.endColumn!);
-                    range = Range.create(position, position);
+                    if(!isNaN(token.startOffset)) {
+                        const position = Position.create(token.endLine! - 1, token.endColumn!);
+                        range = Range.create(position, position);
+                    } else {
+                        // No valid prev token. Might be empty document or containing only hidden tokens.
+                        // Point to document start
+                        range = Range.create(0,0,0,0);
```
Comment (msujew) [inline packages/langium/src/validation/document-validator.ts:85]: Nitpick: 
```suggestion
                        range = Range.create(0, 0, 0, 0);
```

File: packages/langium/test/validation/document-validator.test.ts [hunk 2]
```diff
@@ -46,4 +46,11 @@ describe('Parser error is thrown on resynced token with NaN position', () => {
         const endPosition = Position.create(4, 13);
         expect(diagnostics[0].range).toStrictEqual(Range.create(endPosition, endPosition));
     });
+
+    test('Diagnostic is shown at document start when document is empty', async () => {
+        const validationResult = await validate('');
+        const diagnostics = validationResult.diagnostics;
+        expect(diagnostics).toHaveLength(1);
+        expect(diagnostics[0].range).toStrictEqual(Range.create(0,0,0,0));
```
Comment (msujew) [inline packages/langium/test/validation/document-validator.test.ts:54]: Nitpick:
```suggestion
        expect(diagnostics[0].range).toStrictEqual(Range.create(0, 0, 0, 0));
```

**Review by msujew — APPROVED** [review]
LGTM 👍

**Review by spoenemann — COMMENTED** [review]
WDYT about making it a `while` loop following `previousToken` until either a valid range is found or we hit the start of the doc?

**Comment by dhuebner** [comment]
@spoenemann 
Yes, that was my first idea, but as I started working on this issue, I was not able to reproduce the issue as described here https://github.com/langium/langium/issues/571#issuecomment-1178649922 

With am empty document example I only see `previousToken ` on a parserError object, but no `previousToken ` property on the token itself

**Pull request merged**